orchagent/orch.cpp

```diff
@@ -214,14 +215,22 @@ size_t Consumer::refillToSync()
         } while (update_size != 0);
         return total_size;
     }
-    else
+    else if (consumertable != NULL)
     {
         // consumerTable is either ConsumerStateTable or ConsumerTable
-        auto db = getDbConnector();
+        auto db = consumertable->getDbConnector();
         string tableName = getTableName();
         auto table = Table(db, tableName);
         return refillToSync(&table);
     }
+    else
+    {
+        // The consumerTable should be ZmqConsumerStateTable.
+        DBConnector db("APPL_DB", 0);
```

Comment:
The ZmqConsumer is designed for Dash scenario, we still under discussion about if we need put the APPL_DB on remote side. if the DB on host, then code here need connect to a remote Redis instance.

I confirmed that the Dash has no plan to support warm-reboot, so we may need wait Dash finalize design then decide how to implement the refill.